src/corelibs/U2Core/src/globals/GUrl.cpp

```diff
@@ -153,13 +153,13 @@ QByteArray GUrl::getURLStringAnsi(int codePage) const {
 
     DWORD buffSize = WideCharToMultiByte(codePage, 0, wPath.c_str(), -1, NULL, 0, NULL, NULL);
     if (!buffSize) {
-        return nullptr;
+        return QByteArray();
```

Comment:
Your original code was 100% correct. 
QByteArray(nullptr) is equal to the QByteArray()

Docs:
https://doc.qt.io/qt-5/qbytearray.html#QByteArray-1

Example:
    QByteArray nullArray(nullptr);
    printf("Data '%s'\n", nullArray.constData());
    fflush(stdout);

Reply:
It is much more better to use default constructor instead of implicit conversion.